File: src/RehabEasy.Domain/Models/RehabEasyRecord.cs
namespace RehabEasy.Domain.Models;

public sealed class RehabEasyRecord
{
    public string Id { get; init; } = string.Empty;
    public string SourceId { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public string Sender { get; init; } = string.Empty;
    public string Recipient { get; init; } = string.Empty;
    public DateTimeOffset ReceivedAt { get; init; }
    public string Summary { get; init; } = string.Empty;
    public string PlainTextContent { get; init; } = string.Empty;
    public string HtmlContent { get; init; } = string.Empty;
    public IReadOnlyList<string> Tags { get; init; } = Array.Empty<string>();
    public string RawPayloadJson { get; init; } = "{}";
    public string PatientId { get; init; } = string.Empty;
    public string TestType { get; init; } = string.Empty;
    public string PdfLocalPath { get; init; } = string.Empty;
}

File: java/src/main/java/com/rehabeasy/model/RehabEasyRecord.java
package com.rehabeasy.model;

import java.time.Instant;
import java.util.List;

public record RehabEasyRecord(
        String id,
        String sourceId,
        String title,
        String sender,
        String recipient,
        Instant receivedAt,
        String summary,
        String plainTextContent,
        String htmlContent,
        List<String> tags,
        String rawPayloadJson,
        String patientId,
        String testType,
        String pdfLocalPath
) {
    public RehabEasyRecord {
        id = safe(id);
        sourceId = safe(sourceId);
        title = safe(title);
        sender = safe(sender);
        recipient = safe(recipient);
        receivedAt = receivedAt == null ? Instant.EPOCH : receivedAt;
        summary = safe(summary);
        plainTextContent = safe(plainTextContent);
        htmlContent = safe(htmlContent);
        tags = tags == null ? List.of() : List.copyOf(tags);
        rawPayloadJson = rawPayloadJson == null || rawPayloadJson.isBlank() ? "{}" : rawPayloadJson;
        patientId = safe(patientId);
        testType = safe(testType);
        pdfLocalPath = safe(pdfLocalPath);
    }

    private static String safe(String value) {
        return value == null ? "" : value;
    }
}
